### PR TITLE
fix drains for add-on drains

### PR DESCRIPTION
**Pull request by dickeyxxx**

### CODE REVIEW

**Comment by dickeyxxx** [comment]
@bjeanes

File: commands/drains/index.js [hunk 1]
```diff
@@ -10,11 +11,22 @@ function* run (context, heroku) {
...
+      for (let drain of drains[1]) {
+        let output = `${cli.color.cyan(drain.url)} (${cli.color.green(drain.token)})`;
+        if (drain.extended) output = output + ` drain_id=${drain.extended.drain_id}`;
+        cli.log(output);
+      }
+    }
+    if (drains[0].length > 0) {
+      cli.styledHeader('Add-on Drains');
+      for (let drain of drains[0]) {
+        let addon = yield heroku.get(`/apps/${context.app}/addons/${drain.addon.name}`);
```
Comment (bjeanes) [inline commands/drains/index.js]: Will it serialise or parallelise these calls?

**Comment by bjeanes** [comment]
Nice. I like it

File: commands/drains/index.js (hunk 1 above)
Comment (dickeyxxx) [inline commands/drains/index.js]: it'll run in serial, printing them out as it goes

Comment (bjeanes) [inline commands/drains/index.js]: I believe doing `addons = yield drains.map(function(drain) { return heroku.get('/apps/...); })` should perform the requests in parallel at least.

Comment (dickeyxxx) [inline commands/drains/index.js]: that would get part of the way there, I would have to do an `addons.find()` to lookup the add-on later though right?

Comment (dickeyxxx) [inline commands/drains/index.js]: alright made it parallel

Comment (bjeanes) [inline commands/drains/index.js]: > I would have to do an addons.find() to lookup the add-on later though right?

Or [`zip`](http://underscorejs.org/#zip) the two arrays together

**Comment by bjeanes** [comment]
@dickeyxxx this is awesome. Just tested it and it works locally great:

terminal -zsh 225x81 2016-03-30 15-51-50

:shipit:

**Pull request merged**